framework/src/service/ServiceRegistrationCoreInfo.h

```diff
@@ -0,0 +1,100 @@
+/*=============================================================================
+
+  Library: CppMicroServices
+
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+=============================================================================*/
+
+#ifndef CPPMICROSERVICES_SERVICEREGISTRATIONCOREINFO_H
+#define CPPMICROSERVICES_SERVICEREGISTRATIONCOREINFO_H
+
+#include "cppmicroservices/ServiceInterface.h"
+#include "cppmicroservices/ServiceReference.h"
+#include "cppmicroservices/detail/Threads.h"
+
+#include "Properties.h"
+#include "BundlePrivate.h"
+
+#include <atomic>
+
+namespace cppmicroservices
+{
+
+    /**
+     * \ingroup MicroServices
+     */
+    class ServiceRegistrationCoreInfo : public detail::MultiThreaded<>
+    {
+
+      public:
+        ServiceRegistrationCoreInfo(BundlePrivate* bundle_, InterfaceMapConstPtr service, Properties&& props);
+        ~ServiceRegistrationCoreInfo();
```

Comment:
`= default;` can be done here instead of in the .cpp file

Reply:
fixed in https://github.com/CppMicroServices/CppMicroServices/pull/841/commits/446305785ce7eb1d76ffcbf05f3e5342602c3cf7